WIP Add pam_tty_audit.so to sudo, login pam services

mkg20001 opened the pull request:
Fixes PL-129625

~~Currently it doesn't seem to log anything and `aureport --tty ` is only complaining about missing files~~

~~I suspect something needs to be enabled for this to work?~~

@flyingcircusio/release-managers

## Release process

Impact:

Changelog:

## Security implications

- [ ] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
- [ ] Security requirements tested? (EVIDENCE)

ctheune commented on nixos/infrastructure/testing.nix:
```diff
@@ -18,11 +18,19 @@
     # build-vms.nix from NixOS automatically generates numbered interface
     # configs with default IPs. We rename the devices to fe and srv early so
-    # the services wait for eth1 and eth2 for 5 minutes and time out.
+    # the services wait for the interfaces for 5 minutes and time out.
```
I like this! Can we extract that into a separate PR, please?

mkg20001 replied:
Has been split https://github.com/flyingcircusio/fc-nixos/pull/372

mkg20001 replied:
Note that I'd need that other PR merged before this one then

ctheune replied:
Done

ctheune commented on versions.json:
```diff
@@ -1,9 +1,9 @@
 {
   "nixpkgs": {
-    "owner": "NixOS",
+    "owner": "mkg20001",
```
We need to clean this up before merging!

mkg20001 replied:
upstream PR for this is https://github.com/NixOS/nixpkgs/pull/137661

ctheune commented on tests/testlib.nix:
```diff
@@ -1,5 +1,26 @@
 { lib }:
-{
+
```
This sounds good, I'd like if we could merge that together with the separate PR from the first change in here.

mkg20001 committed:
testlib: network & config template, test keys

- A SSH testkey is now provided as testkey.priv/pub
- Network configuration template is provided, documentation in 
testlib.nix

mkg20001 committed:
elasticsearch: switch to new testlib

ctheune committed:
Merge pull request #372 from mkg20001/testlib

testlib: network & config template, test keys

mkg20001 committed:
audit: add audit & auditbeat

Changelog:

- Add pam_tty_audit.so to sshd, login pam services
- Enable audit system (auditd, audit module)
- Add audit logging test for pam_tty_audit
- Added auditbeat module
- Added filebeat module and made auditbeat upload via filebeat

ctheune commented:
I'm merging this onto a branch to make a few adjustments.
